components/playlist/browser/resources/reducers/playlist_reducer.ts

```diff
@@ -3,48 +3,51 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-import { Playlist } from 'components/definitions/playlist'
+import * as States from 'components/playlist/browser/resources/reducers/states'
 import { Reducer } from 'redux'
 
 import { types } from '../constants/playlist_types'
 
 import * as PlaylistMojo from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m.js'
 
-const playlistReducer: Reducer<Playlist.State | undefined> =
-    (state: Playlist.State | undefined, action) => {
-      if (state === undefined) {
-        state =
-          { lists: [], currentList: undefined, lastPlayerState: undefined }
+const playlistReducer: Reducer<States.PlaylistData | undefined> = (
+  state: States.PlaylistData | undefined,
+  action
+) => {
+  if (state === undefined) {
+    state = { lists: [], currentList: undefined, lastPlayerState: undefined }
+  }
+
+  switch (action.type) {
+    case types.PLAYLIST_LOADED:
+      const playlists = action.payload
+      let currentList: PlaylistMojo.Playlist | undefined
+      if (state.currentList) {
+        currentList = playlists.find((list: PlaylistMojo.Playlist) => {
+          return list.id === state?.currentList?.id
+        })
       }
+      if (!currentList) currentList = playlists[0]
 
-      switch (action.type) {
-        case types.PLAYLIST_LOADED:
-          const playlists = action.payload
-          let currentList: PlaylistMojo.Playlist | undefined
-          if (state.currentList) {
-            currentList = playlists.find((list: PlaylistMojo.Playlist) => {
-              return list.id === state?.currentList?.id
-            })
-          }
-          if (!currentList) currentList = playlists[0]
-
-          if (playlists.length && !currentList) {
-            console.error('there\'s no selected playlist even though we have playlists')
-          }
-
-          state = { ...state, currentList, lists: [...playlists] }
-          break
-
-        case types.PLAYLIST_SELECTED:
-          const playlist = action.payload
-          state = { ...state, currentList: playlist }
-          break
-
-        case types.PLAYLIST_PLAYER_STATE_CHANGED:
-          state = { ...state, lastPlayerState: action.payload }
-          break
+      if (playlists.length && !currentList) {
```

Comment:
With the previous line, I don't think this can ever happen, so it's probably safe to remove.